backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/process/impl/ProcessRestController.java

```diff
@@ -0,0 +1,232 @@
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping(ProcessRestController.ENDPOINT)
+@Profile(Profiles.PROFILE_App)
+public class ProcessRestController
+{
+	private static final transient Logger logger = LogManager.getLogger(ADProcessDAO.class);
+
+	public static final String ENDPOINT = MetasfreshRestAPIConstants.ENDPOINT_API + "/process";
+
+	private final IADProcessDAO adProcessDAO = Services.get(IADProcessDAO.class);
+	private final PermissionServiceFactory permissionServiceFactory = PermissionServiceFactories.currentContext();
+
+	private final ProcessService processService;
+
+	public ProcessRestController(final ProcessService processService)
+	{
+		this.processService = processService;
+	}
+
+	@PostMapping("{value}/invoke")
+	public ResponseEntity<?> invokeProcess(
+			@NonNull @PathVariable("value") final String processValue,
+			@Nullable @RequestBody final RunProcessRequest request)
+	{
+
+		final Optional<AdProcessId> processId = getProcessIdIfRunnable(processValue);
+
+		if (!processId.isPresent())
+		{
+			return ResponseEntity.badRequest().body(Message.of("The requested process is not runnable or it doesn't exist!"));
+		}
+
+		final boolean cannotRunProcess = !permissionServiceFactory.createPermissionService().canRunProcess(processId.get());
+
+		if (cannotRunProcess)
+		{
+			return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+		}
+
+		final ProcessInfo.ProcessInfoBuilder processInfoBuilder = ProcessInfo.builder();
+		processInfoBuilder.setAD_Process_ID(processId.get());
+
+		final boolean parametersPresent = request != null && !Check.isEmpty(request.getProcessParameters());
+
+		if (parametersPresent)
+		{
+			request.getProcessParameters().forEach(param -> processInfoBuilder.addParameter(param.getName(), param.getValue()));
+		}
+
+		final ProcessExecutionResult processExecutionResult = processInfoBuilder
+				.buildAndPrepareExecution()
+				.executeSync()
+				.getResult();
+
+		return getResponse(processExecutionResult);
+	}
+
+	@GetMapping("/available")
+	public ResponseEntity<GetAvailableProcessesResponse> getAllProcesses()
+	{
+		final List<ProcessBasicInfo> processList = processService.getProcessesByType(ProcessType.getTypesRunnableFromAppRestController());
+
+		final List<JSONProcessBasicInfo> jsonProcessList = processList
+				.stream()
+				.map(this::buildJSONProcessBasicInfo)
+				.collect(Collectors.toList());
+
+		final GetAvailableProcessesResponse response = GetAvailableProcessesResponse
+				.builder()
+				.availableProcesses(jsonProcessList)
+				.build();
+
+		return ResponseEntity
+				.ok()
+				.contentType(MediaType.APPLICATION_JSON_UTF8)
+				.body(response);
+	}
+
+	private ResponseEntity<?> getResponse(@NonNull final ProcessExecutionResult processExecutionResult)
+	{
+		if (processExecutionResult.isError())
+		{
+			return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(Message.of("Ups! Looks like something went wrong on our end..."));
```

Comment:
i'm not sure its great to send the whole stacktrace or if that's too much..
but on the other hand, here i think at least the exception class and message should be shown..and maybe, also the whole stacktrace after all (check e.g. the olcand-record).